### PR TITLE
A couple of improvements to user handling

### LINKED ISSUE #180: Username should always be immutable

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
Closes #180

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ class User < ApplicationRecord # rubocop:disable Metrics/ClassLength
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :registerable, :recoverable, :rememberable, :validatable,
-         :fido_usf_registerable
+         :fido_usf_registerable, :timeoutable
```
Comment (igalic) [inline app/models/user.rb:38]: and this is all that's necessary??

**Review by igalic — APPROVED** [review]
looks pretty good.

a lot of the issues we have with usernames would be solved if we used UIDs for computers and usernames for humans… alas…

File: app/models/user.rb (hunk 1 above)
Comment (ChrisMacNaughton) [reply app/models/user.rb:38]: yup, telling Devise to time out a model is all that's needed to make user sessions expire after inactivity :)

**Comment by ChrisMacNaughton** [comment]
Part of the problem is that Service Providers depend on a field being consistent and generally expect it to be either a username or email (and it's also usually configurable as to which is immutable), rather than allowing a GUID, or other opaque identifier

**Pull request merged**